system/jlib/jsocket.cpp

```diff
@@ -4734,7 +4741,10 @@ class CSocketEpollThread: public CSocketBaseThread
         validateerrcount = 0;
         offset = 0;
         selecttrace = trc;
-        epfd = ::epoll_create(MAX_RET_EVENTS);
+        hdlPerThrd = _hdlPerThrd;
+        if (hdlPerThrd == 0)
+            hdlPerThrd = epoll_hdlPerThrd;
```

Comment:
minor: the defaults are a little confusing IMO - use of 0 vs -1 and multiple checks.
CSocketEpollThread in internal and doesn't need a default param (and isn't used). I would remove the default and the check above and make sure a valid value is passed in (which you already do).

Reply:
we generally shy away from defaults params, they can be confusing, especially when the code is extended/extra prams added - better to be explicit where possible.